packages/flutter/lib/src/widgets/scroll_context.dart

```diff
@@ -8,11 +8,34 @@ import 'package:flutter/rendering.dart';
 import 'framework.dart';
 import 'ticker_provider.dart';
 
+/// An interface that [Scrollable] widgets implement in order to use
+/// [ScrollPosition].
+///
+/// See also:
+///
+///  * [ScrollableState], which is the most common implementation of this
+///    interface.
+///  * [ScrollPosition], which uses this interface to communicate with the
+///    scrollable widget.
 abstract class ScrollContext {
+  /// The [BuildContext] that should be used when dispatching
+  /// [ScrollNotification]s.
+  ///
+  /// This context is typically different that the context of the scrollable
+  /// widget itself. For example, [Scrollable] uses a context outside the
+  /// [Viewport] but inside the widgets created by
+  /// [ScrollBehavior.buildViewportChrome].
   BuildContext get notificationContext;
+
+  /// A [TickerProvider] to use when animating the scroll position.
   TickerProvider get vsync;
+
+  /// The direction in which the widget scrolls.
   AxisDirection get axisDirection;
 
+  /// Whether the widget should ignore [PointerEvent] inputs.
```

Comment:
Specifically this is about whether the entire widget subtree should ignore pointers. It makes the contents of the widget non-interactive.